src/modules/extra/m_mysql.cpp

```diff
@@ -255,6 +255,15 @@ class SQLConnection : public SQLProvider
 		bool rv = mysql_real_connect(connection, host.c_str(), user.c_str(), pass.c_str(), dbname.c_str(), port, NULL, 0);
 		if (!rv)
 			return rv;
+		// enable character-set settings
+		std::string characterset;
+		if (config->readString("characterset", characterset))
```

Comment:
Are you deliberately using `readString()` to allow empty strings? If not, please use `getString()` as the code above and test for empty() to determine whether or not to call the mysql_ func.